tensorflow/lite/delegates/gpu/cl/egl_sync.cc

```diff
@@ -21,9 +21,49 @@ namespace tflite {
 namespace gpu {
 namespace cl {
 
+namespace {
+
+PFNEGLCREATESYNCKHRPROC g_eglCreateSyncKHR = nullptr;
```

Comment:
i have a couple issues with these:
1. global variables, and 
2. no mutex

How often is IsEglSyncSupported called and how slow is eglGetProcAddress?  If it's rare and not performance critical, I would just make it simple without global variables.

Reply:
I've tried to basically replicate the code from gl_interop.cc: https://github.com/tensorflow/tensorflow/blob/b780f710ce79cdabef13da19608c80f83f45141c/tensorflow/lite/delegates/gpu/cl/gl_interop.cc#L63-L74
As far as I understand, static initializers are guaranteed to be thread-safe since C++11, though I may be misunderstanding how that deals with the side effects.
`eglGetProcAddress` may have a non-trivial run time, depending on the vendor. In any case, loading functions should probably be performed at initialization time, so maybe I should move this to `opencl_wrapper.cc/.h`?

Reply:
Ah missed the static there.  ok.